pkg/grpc/server.go

```diff
@@ -147,6 +152,16 @@ func (a *apiImpl) Register(services ...APIService) {
 	a.apiServices = append(a.apiServices, services...)
 }
 
+func (a *apiImpl) Stop() *concurrency.Signal {
+	if a.grpcShutdownRequested.IsDone() {
+		log.Warnf("API Stop called but shutdown was already requested")
```

Comment:
Why is here a warning being logged if it is okay to call `Stop` multiple times?
I would expect as a caller to either have an error returned or always receive the signal.

Reply:
Fixed